source/SemVerAnalyzer/Nuget/NugetClient.cs

```diff
@@ -21,7 +25,32 @@ public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, string
 			try
 			{
 				using var client = new HttpClient();
-				using var request = new HttpRequestMessage(HttpMethod.Get, Path.Join(_config.RepositoryUrl, packageName));
+				
+				using var feedRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
+				using var feedResponse = await client.SendAsync(feedRequest);
+				if (!feedResponse.IsSuccessStatusCode)
+				{
+					var content = await feedResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving Nuget feed:\n{content}");
+					return null;
+				}
+
+				var feed = JsonSerializer.Deserialize<NugetPackageFeed>(await feedResponse.Content.ReadAsStringAsync());
+				var packageBaseAddress = feed.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
```

Comment:
I'm concerned about reading from the feed without any kind of filtering.  We want to go directly to the package, not read a (paged) list of _all_ of the packages available.  What if our package source is nuget.org?  There are millions of packages.  Are you going to page through them all to find the one you're looking for?

Reply:
This does not read a list of packages, just the nuget API metadata. Nuget's own one is here:
https://api.nuget.org/v3/index.json